src/_guides/language/extension-methods.md

```diff
@@ -0,0 +1,246 @@
+
+{% comment %}
+[PENDING: embed video here.]
+Watch this video for an introduction to extension methods.
+{% endcomment %}
+
+## Overview
+
+When you’re using someone else’s API or
+when you implement a library that’s widely used,
+it’s often impractical or impossible to change the API.
+But you might still want to add some functionality.
+
+For example, consider the following code that parses a string into an integer:
+
+```dart
+int.parse('42')
+```
+
+It might be nice — shorter and easier to use with tools — to
+have that functionality be in the string instead:
+
+```dart
+'42'.toInt()
+```
+
+To enable that code,
+you can import a library that contains an extension of the String class:
+
+```dart
+import string_apis.dart';
+...
+print('42'.toInt());
+```
+
+Extensions can define not just methods and getters,
+but also other members such as setters and operators.
+Also, extensions have names, which can be helpful if an API conflict arises.
+Here's how you might implement the extension method `toInt()`,
+using an extension (named `ParseNumbers`) that operates on strings:
+
+```dart
+extension ParseNumbers on String {
+  int toInt() {
+    return int.parse(this);
+  }
+}
+```
+
+The next section describes how to _use_ extension methods.
+After that are sections about _implementing_ extension methods.
+
+
+## Using extension methods
+
+Like all Dart code, extension methods are in libraries.
+You’ve already seen how to use an extension method —
+just import the library it’s in, and use it like an ordinary method:
+
+```dart
+// Import a library that contains an extension on String. 
+import string_apis.dart';
+...
+print('42'.padLeft(5)); // Use a String method.
+print('42'.toInt()); // Use an extension method.
+```
+
+That’s all you usually need to know to use extension methods.
+As you write your code, you might also need to know
+how extension methods depend on static types (as opposed to `dynamic`) and
+how to resolve [API conflicts](#api-conflicts).
+
+### Static types and dynamic
+
+You can’t invoke extension methods on variables of type `dynamic`.
+For example, the following code results in a runtime exception:
+
+```dart
+dynamic d = '2';
+print(d.toInt()); // Runtime exception: NoSuchMethodError
+```
+
+Extension methods _do_ work with Dart’s type inference.
+The following code is fine because
+the variable `v` is inferred to have type `String`:
+
+```dart
+var v = '2';
+print(v.toInt()); // Output: 2
+```
+
+The reason that `dynamic` doesn’t work is that
+extension methods are resolved against the static type of the receiver.
+For more information about static types and `dynamic`, see
+[The Dart type system](/guides/language/sound-dart).
+
+### API conflicts
+
+If an extension member conflicts with
+an interface or with another extension member,
+then you have a few options.
+
+One option is changing how you import the conflicting extension.
+You can use `show` or `hide`, or you can remove the import:
```

Comment:
I would remove the "remove the import" part. That seems a bit given...

Reply:
I've rewritten it slightly.

Reply:
Proposed:

> One option is changing how you import the conflicting extension,
using `show` or `hide` to limit the exposed API: